homeassistant/components/tradfri/sensor.py

```diff
@@ -23,38 +24,48 @@ async def async_setup_entry(
 ) -> None:
     """Set up a Tradfri config entry."""
     gateway_id = config_entry.data[CONF_GATEWAY_ID]
-    tradfri_data = hass.data[DOMAIN][config_entry.entry_id]
-    api = tradfri_data[KEY_API]
-    devices = tradfri_data[DEVICES]
+    coordinator_data = hass.data[DOMAIN][config_entry.entry_id][COORDINATOR]
+    api = coordinator_data[KEY_API]
 
     async_add_entities(
-        TradfriSensor(dev, api, gateway_id)
-        for dev in devices
+        TradfriSensor(
+            device_coordinator,
+            api,
+            gateway_id,
+        )
+        for device_coordinator in coordinator_data[COORDINATOR_LIST]
         if (
-            not dev.has_light_control
-            and not dev.has_socket_control
-            and not dev.has_blind_control
-            and not dev.has_signal_repeater_control
-            and not dev.has_air_purifier_control
+            not device_coordinator.device.has_light_control
+            and not device_coordinator.device.has_socket_control
+            and not device_coordinator.device.has_blind_control
+            and not device_coordinator.device.has_signal_repeater_control
+            and not device_coordinator.device.has_air_purifier_control
         )
     )
 
 
-class TradfriSensor(TradfriBaseDevice, SensorEntity):
+class TradfriSensor(TradfriBaseEntity, SensorEntity):
     """The platform class required by Home Assistant."""
 
     _attr_device_class = SensorDeviceClass.BATTERY
     _attr_native_unit_of_measurement = PERCENTAGE
 
     def __init__(
         self,
-        device: Command,
+        device_coordinator: TradfriDeviceDataUpdateCoordinator,
         api: Callable[[Command | list[Command]], Any],
         gateway_id: str,
     ) -> None:
-        """Initialize the device."""
-        super().__init__(device, api, gateway_id)
-        self._attr_unique_id = f"{gateway_id}-{device.id}"
+        """Initialize a switch."""
+        super().__init__(
+            device_coordinator=device_coordinator,
+            api=api,
+            gateway_id=gateway_id,
+        )
+
+    def _refresh(self) -> None:
+        """Refresh the device."""
+        self._attr_native_value = self.coordinator.data.device_info.battery_level
 
     @property
     def native_value(self) -> int | None:
```

Comment:
We don't need this property if we set native value attribute above in the refresh.

Reply:
Oh, they're the same? I'll remove it.

Reply:
Had to do an `__init__` refresh for tests to pass though.